course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -99,14 +146,129 @@
+            if (scrollToAlerts || scrollToAlerts == null) {
+                scrollTo(0, 0);
+            }
+        };
+
+        dc.submitCourseDetailsForm = function() {
+            dc.courseDetailsUpdateInProgress = true;
+            var postData = {};
+            var url = djangoUrl.reverse(dc.apiProxy,
+                [dc.apiBase + dc.courseInstanceId + '/']);
+            // we could also get these from editable properties on the DOM
+            var fields = [
+                'description',
+                'instructors_display',
+                'location',
+                'meeting_time',
+                'notes',
+                'short_title',
+                'sub_title',
+                'title'
+            ];
+            fields.forEach(function(field) {
+                postData[field] = dc.formDisplayData[field];
+            });
+
+            $http.patch(url, postData)
+                .then(function finalizeCourseDetailsPatch() {
+                    // update Reset button
+                    $.extend(dc.courseInstance, postData);
+                    dc.resetGlobalAlerts();
+                    dc.alerts.global.updateSucceeded = {show: true};
+                }, function cleanUpFailedCourseDetailsPatch(response) {
+                    dc.handleAjaxErrorResponse(response);
+                    dc.resetGlobalAlerts();
+                    dc.alerts.global.updateFailed = {
+                        show: true,
+                        details: response.statusText || 'None'};
+                })
+                .finally( function courseDetailsUpdateNoLongerInProgress() {
+                    dc.courseDetailsUpdateInProgress = false;
+                });
+        };
+
+        dc.init();
+    }
+
+    function editableInputDirective() {
+        return {
+            scope: {
+                editable: '=', // can be < in angular 1.5;
+                               // if not provided, defaults to null/false
+                field: '@',
+                formValue: '=',
+                isLoading: '&',
+                label: '@',
+                maxlength: '@',
+                modelValue: '='
+            },
+            template: ' \
+<li class="list-group-item"> \
+  <div class="form-group"> \
+    <label for="input-course-{{field}}" class="col-md-2"> \
+      {{label}} \
+    </label> \
+    <div class="col-md-10"> \
+      <span ng-show="isLoading()"><i class="fa fa-refresh fa-spin"></i></span> \
+      <div ng-hide="isLoading()"> \
+        <input type="text" class="form-control" id="input-course-{{field}}" \
+               name="input-course-{{field}}" ng-show="editable" \
+               ng-model="formValue" maxlength="{{maxlength}}"/> \
+        <span ng-hide="editable">{{modelValue}}</span> \
+      </div> \
+    </div> \
+  </div> \
+</li> \
+'
+        }
     }
+
+    function fieldLabelWrapperDirective() {
+        return {
+            //templateUrl: 'directives/field_label_wrapper.html'
```

Comment:
dev cruft.

fwiw, i'm guessing the issue was the relative path here is relative to the url of the page, so you'd need to prefix it with `'static/js/'`.  that being said, i'd put it in partials, not in the `js/directives` subdir.